backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,339 @@
+        this.backtraceApi = backtraceApi;
+    }
+
+    /**
+     * Enable metrics
+     *
+     * @param settings for Backtrace metrics
+     */
+    public void enable(BacktraceMetricsSettings settings) {
+        this.settings = settings;
+        // For tracking crash-free sessions we need to add
+        // application.session and application.version to Backtrace attributes
+        String sessionId = UUID.randomUUID().toString();
+        this.customReportAttributes.put("application.session", sessionId);
+        try {
+            this.customReportAttributes.put("application.version", this.context.getPackageManager()
+                    .getPackageInfo(this.context.getPackageName(), 0).versionName);
+        } catch (PackageManager.NameNotFoundException e) {
+            BacktraceLogger.e(LOG_TAG, "Could not resolve package version information for Backtrace metrics");
+        }
+
+        try {
+            startMetricsEventHandlers(backtraceApi);
+            sendStartupEvent();
+        } catch (Exception e) {
+            BacktraceLogger.e(LOG_TAG, "Could not enable metrics, exception " + e.getMessage());
+        }
+    }
+
+    private void startMetricsEventHandlers(Api backtraceApi) {
+        uniqueEventsHandler = backtraceApi.enableUniqueEvents(context, customReportAttributes, settings);
+        summedEventsHandler = backtraceApi.enableSummedEvents(context, customReportAttributes, settings);
+    }
+
+    protected String getStartupUniqueEventName() {
+        return this.startupUniqueEventName;
+    }
+
+    public void setStartupUniqueEventName(String startupUniqueEventName) {
+        this.startupUniqueEventName = startupUniqueEventName;
+    }
+
+    public String getBaseUrl() {
+        return this.settings.getBaseUrl();
+    }
+
+    /**
+     * Send startup event to Backtrace
+     */
+    public void sendStartupEvent() {
+        uniqueEventsHandler.sendStartupEvent(startupUniqueEventName);
+        summedEventsHandler.sendStartupEvent(startupSummedEventName);
+    }
+
+    /**
+     * Send all outgoing messages (unique and summed) currently queued
+     */
+    public void send() {
+        uniqueEventsHandler.send();
+        summedEventsHandler.send();
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     *
+     * @param attributeName Attribute name
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName) {
+        return addUniqueEvent(attributeName, null);
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     *
+     * @param attributeName Attribute name
+     * @param attributes    Event attributes
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName, Map<String, Object> attributes) {
+        if (!shouldProcessEvent(attributeName)) {
+            BacktraceLogger.w(LOG_TAG, "Skipping report: Reached store limit or event has empty name.");
```

Comment:
let's log here attribute name and count() to simplify debugging

Reply:
Good idea, thank you!

Reply:
@BartoszLitwiniuk I will move logging to inside of `shouldProcessEvent`